src/lib.rs

```diff
@@ -160,6 +173,72 @@ impl L402Module {
             }
         }
     }
+
+    pub async fn verify_cashu_token(&self, token: &str, amount_msat: i64) -> Result<bool, String> {
+        // Decode the token from string
+        let token_decoded = match cdk::nuts::Token::from_str(token) {
+            Ok(token) => token,
+            Err(e) => {
+                eprintln!("Failed to decode Cashu token: {}", e);
+                return Err(format!("Failed to decode Cashu token: {}", e));
+            }
+        };
+        
+        // Check if the token is valid
+        if token_decoded.proofs().is_empty() {
+            return Ok(false);
+        }
+        
+        // Calculate total token amount in millisatoshis
+        let total_amount_msat: u64 = token_decoded.proofs().iter()
+            .map(|p| u64::from(p.amount) * MSAT_PER_SAT)
+            .sum();

```

Comment:
Could use `token_decoded.value()` and then convert to msats https://docs.rs/cashu/latest/cashu/nuts/nut00/token/struct.TokenV4.html#method.value